namerd/iface/interpreter-thrift-idl/src/main/thrift/namer.thrift

```diff
@@ -81,6 +81,7 @@ struct AddrReq {
 }
 
 struct AddrMeta {
+  1: map<string, string> fields
```

Comment:
I'd prefer to enumerate the set of supported metadata, until we really need an arbitrary map.

Let's have an optional `authority` field (to match HTTP/2 conventions) that captures what we're doing here.